### DISCUSSION TITLE
Document editor

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#68] [Ideas]
What is your plan with document editor?
An universal document viewer able to convert to a BleskOS document format for editing and saving?

**Comment by VendelinSlezak** [comment]
Document editor is BleskOS equivalent of programs like Microsoft Word or Libreoffice Writer. That’s of course really big thing, so realistic plan in near future is that it will be able to open and save to ODT and DOCX with support for:

- different font size
- font emphasis
- font color
- font background color
- paragraphs
- alignment of paragraphs
- space around paragraphs
- lists
- line breaks
- pages of different sizes
- page breaks

BleskOS document memory format is only virtual type of storing document data that is suitable for real-time editing.

This is how this whole idea works:
Opening file: ODT, DOC, DOCX, PDF… => BleskOS document memory format => BleskOS low level document memory format
Showing file on screen: BleskOS low level memory format - it very fast, so even document with hundreds of pages will have smooth scrolling
Editing file: changes goes to BleskOS document memory format => changes recalculated into BleskOS low level document memory format
Saving file: BleskOS document memory format => ODT/DOCX

When I will reach this goal, I will move on Spreadsheet editor.

**Comment by defdefred** [comment]
Very big things indeed.
I like the viewer feature.
I like the fast internal format, must be able to save to this format.
Not sure of the usefulness of saving to other format as it is always a mess of incompatibility when working the same document with different software (except for PDF who is a printing output format readable everywhere).

I see the world going to modern format as markdown or html and less stuck with ancient tool made primary for printing on paper purpose. Now document need to match with different screen size and resolution + animated image... So you see what I mean: odt and docx are obsolete.

**Comment by VendelinSlezak** [comment]
Yes, I can include option to save BleskOS document memory format, it is much simpler and smaller than ODT or DOCX. But option to save to those formats is also needed, because no other program will support (at least for now) simple formats used internally by BleskOS.

**Reply by defdefred** [reply]
Could be an external tools to convert from BleskOS to whatever.
